src/containers/shared/components/DomainLink.tsx

```diff
@@ -9,12 +9,27 @@ export interface Props {
 
 const DomainLink = (props: Props) => {
   const { className, decode = false, domain } = props
+
+  // Ensure that "https://" is not added twice
+  const domainWithoutProtocol = domain.replace(/^https?:\/\//, '')
```

Comment:
Wouldn't it be easier to just check if the domain has `"http"` and avoid adding it if it does?

Reply:
### Yes, it would be easier to _just check if the domain has "https://" and avoid adding it if it does._

- ### However, if we go by that method, the **5th test** keeps failing:
The test:
```typescript
  it('handles domain link with domain provided in HEX-encoded format', () => {
    const wrapper = mount(
      <DomainLink decode domain="68747470733A2F2F6578616D706C652E636F6D" />,
    )
    expect(wrapper.find('a').props().className).toEqual('domain')
    expect(wrapper.find('a').text()).toEqual('https://example.com')
    expect(wrapper.find('a').props().href).toEqual('https://example.com')
    wrapper.unmount()
  })
```

- ### Feedback / test output after running the test (after implementing - _check if it already has domain and avoid adding one if it does_):
The output from the tests/console:
```
● DomainLink › handles domain link with domain provided in HEX-encoded format

    expect(received).toEqual(expected) // deep equality

    Expected: "https://example.com"
    Received: "https://https://example.com"
```

I attempted to implement your suggested approach, but encountered a persisting issue with the **5th test** failing. I found that only when I include the code for _Ensure that "https://" is not added twice_ does the **5th test** pass consistently, without causing any further complications or errors.

- ### _Ensure that "https://" is not added twice:_
The code:
```typescript
  // Ensure that "https://" is not added twice
  const domainWithoutProtocol = domain.replace(/^https?:\/\//, '')
```

Reply:
I was imagining something like this:
```js
const decodedDomain = decode ? decodeHex(domain) : domain
const domainHasProtocol = decodedDomain.match((/^https?:\/\//)
const href = domainHasProtocol ? decodedDomain : `https://${decodedDomain}`
```

Reply:
One that would include all protocols would be `/^([a-z][a-z0-9+\-.]*):\/\//`